### DISCUSSION TITLE
Unable to get Solito monorepo to work with nativewind

### DISCUSSION THREAD

**Discussion by adherb** [nativewind/nativewind#161] [Q&A] [answered]
I am unable to get my Solito mono repo with nativewind. I have looked at the ```with-nativewind``` repo and even though I have copied everything exactly the same, it will still not work.

Has anyone got this to work before?

**Comment by marklawlor** [comment]
Hi @adherb, can you elaborate on your issues please. Do you get an error or are you able to share a screenshot? I'm also happy to help debug if your able to share your repository.

There is so a troubleshooting guide on the website. https://www.nativewind.dev/guides/troubleshooting

**Comment by adherb** [comment]
Hey @marklawlor, thanks for the reply.

No I'm not getting an error, the Tailwind styles just won't apply, I've triple checked the 'content' and everything looks good to go.

Here is the repo:
```
https://github.com/adherb/solito-with-nativewind
```

I really appreciate your help!

**Comment by marklawlor** [comment]
To clarify your issue, NativeWind is not working within your _Next.js application_. I was unable to replicate any issue with the Expo app. Please let me know if I missed something.

The issue is that `react-native-web` is not a dependency of the Next.js application and is using the version from expo.

```diff
// apps/expo/package.json
-    "react-native-web": "0.17.1"

// apps/next/package.json
+    "react-native-web": "0.18.7",
```

I'd recommend clearing your node_modules and expo & next cache before testing again

**Reply by adherb** [reply]
Thanks Mark, still no luck unfortunately.

I have done the following:
```
// apps/next/package.json
+    "react-native-web": "0.18.7",
```
Deleted all node_modules folders and then ran:
```
yarn install
```

Then ran:
``` 
yarn cache clean
```

I've also tried deleting the project from my local machine and then re-downloading from Github, but still no luck, do you have any other ideas what it could be?

**Comment by adherb** [comment]
Hi again Mark,

Another update for you.

If I use ```<p>``` rather than ```<P>```, the tailwindcss styles show.

For example, this doesn't work:
```
import { A, H1, P, Text, TextLink } from 'app/design/typography'

  <P className="text-center text-red-500 text-purple">
        Here is a basic starter to show you how you can navigate from one screen
        to another. This screen uses the same code on Next.js and React Native.
  </P>
```

This does work:
```
 <p className="text-center text-red-500 text-purple">
        Here is a basic starter to show you how you can navigate from one screen
        to another. This screen uses the same code on Next.js and React Native.
 </p>
```

Also, when I'm inside ```apps/next``` and run ```yarn build```  I get the following error:

```
$ next build
Failed to compile.

../../packages/app/design/typography.tsx:18:3
Type error: Type '{ accessibilityLevel: number; accessibilityRole: "header"; }' is not assignable to type 'Partial<TextProps & { className?: string | undefined; tw?: string | undefined; baseClassName?: string | undefined; baseTw?: string | undefined; } & RefAttributes<unknown>>'.
  Object literal may only specify known properties, but 'accessibilityLevel' does not exist in type 'Partial<TextProps & { className?: string | undefined; tw?: string | undefined; baseClassName?: string | undefined; baseTw?: string | undefined; } & RefAttributes<unknown>>'. Did you mean to write 'accessibilityLabel'?

  16 | export const H1 = styled(RNText, 'text-3xl font-extrabold my-4')
  17 | H1.defaultProps = {
> 18 |   accessibilityLevel: 1,
     |   ^
  19 |   accessibilityRole: 'header',
  20 | }
  21 | 
error Command failed with exit code 1.
```

If i comment out the accessibility rule I get this:
```
Failed to compile.

../../packages/app/features/user/detail-screen.tsx:12:11
Type error: No overload matches this call.
  Overload 1 of 2, '(props: ViewProps | Readonly<ViewProps>): View', gave the following error.
    Type '{ children: Element[]; className: string; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<View> & Readonly<ViewProps> & Readonly<{ children?: ReactNode; }>'.
      Property 'className' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<View> & Readonly<ViewProps> & Readonly<{ children?: ReactNode; }>'.
  Overload 2 of 2, '(props: ViewProps, context: any): View', gave the following error.
    Type '{ children: Element[]; className: string; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<View> & Readonly<ViewProps> & Readonly<{ children?: ReactNode; }>'.
      Property 'className' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<View> & Readonly<ViewProps> & Readonly<{ children?: ReactNode; }>'.

  10 | 
  11 |   return (
> 12 |     <View className="flex-1 justify-center items-center">
     |           ^
  13 |       <Text className="text-center font-bold mb-4">{`User ID: ${id}`}</Text>
  14 |       <TextLink href="/">👈 Go Home</TextLink>
  15 |     </View>
error Command failed with exit code 1.
```

Hope this helps.

**Comment by marklawlor** [comment, marked as answer]
An example project is now available via `npx create-solito-app@latest my-solito-app -t with-tailwind`

https://github.com/nandorojo/solito/tree/master/example-monorepos/with-tailwind

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
